GabrielDSousa/softdesign-test
# Feature/books

## Description

This pull request introduces enhancements to the Angular frontend in the `feature/books` branch. The primary focus is on implementing Book CRUD functionality and making adjustments to display the weather forecast on the home page.

## Changes Made

1. **Book CRUD Implementation:**
   - Added components, services, and templates for listing, adding, updating, and deleting books.
   - Implemented reactive forms for book data input.

2. **Weather Forecast on Home Page:**
   - Integrated with the weather API to fetch and display the current weather information.
   - Enhanced UI components for a better user experience.

## How to Test

1. Clone this repository and switch to the `feature/books` branch.
2. Follow the instructions in the README to set up and run the Angular frontend.
3. Navigate to the Book section to test the CRUD functionality.
4. Observe the home page to see the weather forecast.

## Screenshots

[Attach relevant screenshots if applicable]

## Related Issues

Closes #123

## Checklist

- [x] Code follows the coding standards.
- [x] Tests have been added, or existing tests have been modified to cover the changes.
